pkg/controllers/management/auth/globalroles/fleetworkspace_binding_handler.go

```diff
@@ -0,0 +1,203 @@
+	rbCache   rbacv1.RoleBindingCache
+	fwCache   mgmtcontroller.FleetWorkspaceCache
+}
+
+func newFleetWorkspaceBindingHandler(management *config.ManagementContext) *fleetWorkspaceBindingHandler {
+	return &fleetWorkspaceBindingHandler{
+		crbClient: management.Wrangler.RBAC.ClusterRoleBinding(),
+		crbCache:  management.Wrangler.RBAC.ClusterRoleBinding().Cache(),
+		grCache:   management.Wrangler.Mgmt.GlobalRole().Cache(),
+		rbClient:  management.Wrangler.RBAC.RoleBinding(),
+		rbCache:   management.Wrangler.RBAC.RoleBinding().Cache(),
+		fwCache:   management.Wrangler.Mgmt.FleetWorkspace().Cache(),
+	}
+}
+
+// ReconcileFleetWorkspacePermissionsBindings reconciles backing RoleBindings and ClusterRoleBindings created for granting permission
+// to fleet workspaces.
+func (h *fleetWorkspaceBindingHandler) reconcileFleetWorkspacePermissionsBindings(globalRoleBinding *v3.GlobalRoleBinding) error {
+	globalRole, err := h.grCache.Get(globalRoleBinding.GlobalRoleName)
+	if err != nil {
+		return fmt.Errorf("unable to get globalRole: %w", err)
+	}
+
+	if err = h.reconcileResourceRulesBindings(globalRoleBinding, globalRole); err != nil {
+		return fmt.Errorf("error reconciling fleet permissions rules: %w", err)
+	}
+
+	if err = h.reconcileWorkspaceVerbsBindings(globalRoleBinding, globalRole); err != nil {
+		return fmt.Errorf("error reconciling fleet workspace verbs: %w", err)
+	}
+
+	return nil
+}
+
+func (h *fleetWorkspaceBindingHandler) reconcileResourceRulesBindings(grb *v3.GlobalRoleBinding, gr *v3.GlobalRole) error {
+	fleetWorkspaces, err := h.fwCache.List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("unable to list fleetWorkspaces when reconciling globalRoleBinding %s: %w", grb.Name, err)
+	}
+
+	var returnError error
+	for _, fleetWorkspace := range fleetWorkspaces {
+		if fleetWorkspace.Name == localFleetWorkspace {
+			continue
+		}
+		desiredRB := backingRoleBinding(grb, gr, fleetWorkspace.Name)
+		rb, err := h.rbCache.Get(fleetWorkspace.Name, wrangler.SafeConcatName(grb.Name))
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				returnError = multierror.Append(returnError, err)
+				continue
+			}
+			if gr.InheritedFleetWorkspacePermissions.ResourceRules != nil {
+				_, err = h.rbClient.Create(desiredRB)
+				if err != nil {
+					returnError = multierror.Append(returnError, err)
+				}
+			}
+			continue
+		}
+
+		if gr.InheritedFleetWorkspacePermissions.ResourceRules == nil {
+			err := h.rbClient.Delete(rb.Namespace, rb.Name, &metav1.DeleteOptions{})
+			if err != nil && !apierrors.IsNotFound(err) {
+				returnError = multierror.Append(returnError, err)
+			}
+			continue
+		}
+		if !reflect.DeepEqual(rb.Subjects, desiredRB.Subjects) ||
+			!reflect.DeepEqual(rb.RoleRef, desiredRB.RoleRef) {
+			// undo modifications if rb has changed.
+			err := h.rbClient.Delete(rb.Namespace, rb.Name, &metav1.DeleteOptions{})
+			if err != nil && !apierrors.IsNotFound(err) {
+				returnError = multierror.Append(returnError, err)
```

Comment:
I added a continue after this, so it'll skip creation if deletion fails. Anyway it would have retried in the next reconciliation loop because the err of deletion and the possible creation err would've been returned